instrumentation/faraday/lib/opentelemetry/instrumentation/faraday/middlewares/tracer_middleware.rb

```diff
@@ -42,14 +46,10 @@ def call(env)
           attr_reader :app
 
           def span_creation_attributes(http_method:, url:)
-            instrumentation_attrs = {
-              'http.method' => http_method, 'http.url' => url.to_s
-            }
             config = Faraday::Instrumentation.instance.config
-            instrumentation_attrs['peer.service'] = config[:peer_service] if config[:peer_service]
-            instrumentation_attrs.merge(
-              OpenTelemetry::Common::HTTP::ClientContext.attributes
-            )
+            attributes = from_request(http_method, url, config)
+            attributes['peer.service'] = config[:peer_service] if config[:peer_service]
```

Comment:
`peer.service` feels like something common we should extract as well. It might be more general than just HTTP, though? Maybe we need inheritance of options?

Reply:
I agree and think this can be accomplished via inheritance but as per https://github.com/open-telemetry/opentelemetry-ruby/pull/946#discussion_r723467509 i'm pulling out all the instrumentationoptions stuff into a seperate pr